### PR TITLE
Added API to fetch Kubernetes Objects from subscriber

### PR DESCRIPTION
**Pull request by amityt**
## Proposed changes

This PR adds:
API to get Kubernetes Objects for the following-
1. Pods
2. Deployments
3. Daemonsets
4. Statefulsets
5. DeploymentConfigs
6. Rollouts

- Sample output when deployments data is required
`{
  "data": {
    "getKubeObject": {
      "cluster_id": "a18e3b88-a2e4-4749-9404-a34cb18ca685",
      "kube_obj": "[{\"namespace\":\"default\",\"pods\":null,\"deployments\":[{\"name\":\"nginx-deployment\",\"uid\":\"b5cf2b85-b85c-4dda-8d11-d147a7e1189d\",\"namespace\":\"default\",\"api_version\":\"\",\"cluster_name\":\"\",\"creation_timestamp\":\"2021-03-27T19:57:30Z\",\"replicas\":2,\"node_selectors\":null,\"label_selectors\":{\"matchLabels\":{\"app\":\"nginx\"}},\"labels\":null,\"annotations\":null}],\"statefulsets\":null,\"daemonsets\":null,\"deploymentconfiginfo\":null,\"rollouts\":null},{\"namespace\":\"kube-node-lease\",\"pods\":null,\"deployments\":[],\"statefulsets\":null,\"daemonsets\":null,\"deploymentconfiginfo\":null,\"rollouts\":null},{\"namespace\":\"kube-public\",\"pods\":null,\"deployments\":[],\"statefulsets\":null,\"daemonsets\":null,\"deploymentconfiginfo\":null,\"rollouts\":null},{\"namespace\":\"kube-system\",\"pods\":null,\"deployments\":[{\"name\":\"coredns\",\"uid\":\"5f08f2b2-68ce-49be-b3e8-bdee46a1de87\",\"namespace\":\"kube-system\",\"api_version\":\"\",\"cluster_name\":\"\",\"creation_timestamp\":\"2021-03-27T15:39:12Z\",\"replicas\":1,\"node_selectors\":{\"kubernetes.io/os\":\"linux\"},\"label_selectors\":{\"matchLabels\":{\"k8s-app\":\"kube-dns\"}},\"labels\":{\"k8s-app\":\"kube-dns\"},\"annotations\":null}],\"statefulsets\":null,\"daemonsets\":null,\"deploymentconfiginfo\":null,\"rollouts\":null},{\"namespace\":\"litmus\",\"pods\":null,\"deployments\":[{\"name\":\"argo-server\",\"uid\":\"a4496329-10e5-4d39-b6f2-a06a82cff369\",\"namespace\":\"litmus\",\"api_version\":\"\",\"cluster_name\":\"\",\"creation_timestamp\":\"2021-03-27T17:57:17Z\",\"replicas\":1,\"node_selectors\":{\"kubernetes.io/os\":\"linux\"},\"label_selectors\":{\"matchLabels\":{\"app\":\"argo-server\"}},\"labels\":null,\"annotations\":null},{\"name\":\"chaos-exporter\",\"uid\":\"d0c375f0-690a-42e7-a5bc-ab040dd3c937\",\"namespace\":\"litmus\",\"api_version\":\"\",\"cluster_name\":\"\",\"creation_timestamp\":\"2021-03-27T17:57:18Z\",\"replicas\":1,\"node_selectors\":null,\"label_selectors\":{\"matchLabels\":{\"app\":\"chaos-exporter\"}},\"labels\":{\"app\":\"chaos-exporter\"},\"annotations\":null},{\"name\":\"chaos-operator-ce\",\"uid\":\"7e544cd1-9af3-411e-bac8-9ee02984f670\",\"namespace\":\"litmus\",\"api_version\":\"\",\"cluster_name\":\"\",\"creation_timestamp\":\"2021-03-27T17:57:18Z\",\"replicas\":1,\"node_selectors\":null,\"label_selectors\":{\"matchLabels\":{\"name\":\"chaos-operator\"}},\"labels\":{\"app.kubernetes.io/component\":\"operator\",\"app.kubernetes.io/managed-by\":\"kubectl\",\"app.kubernetes.io/name\":\"litmus\",\"app.kubernetes.io/part-of\":\"litmus\",\"app.kubernetes.io/version\":\"v1.12.0\",\"name\":\"litmus\"},\"annotations\":null},{\"name\":\"event-tracker\",\"uid\":\"d6f78304-eca8-4513-87c1-d04d58dae027\",\"namespace\":\"litmus\",\"api_version\":\"\",\"cluster_name\":\"\",\"creation_timestamp\":\"2021-03-27T17:57:21Z\",\"replicas\":1,\"node_selectors\":null,\"label_selectors\":{\"matchLabels\":{\"app\":\"event-tracker\"}},\"labels\":{\"app\":\"event-tracker\"},\"annotations\":null},{\"name\":\"litmusportal-frontend\",\"uid\":\"bad5456c-5ff2-4b39-b208-c4bb400aaa15\",\"namespace\":\"litmus\",\"api_version\":\"\",\"cluster_name\":\"\",\"creation_timestamp\":\"2021-03-27T17:10:06Z\",\"replicas\":1,\"node_selectors\":null,\"label_selectors\":{\"matchLabels\":{\"component\":\"litmusportal-frontend\"}},\"labels\":{\"component\":\"litmusportal-frontend\"},\"annotations\":null},{\"name\":\"litmusportal-server\",\"uid\":\"3ff6e057-0756-4adb-a3a4-809bafc32142\",\"namespace\":\"litmus\",\"api_version\":\"\",\"cluster_name\":\"\",\"creation_timestamp\":\"2021-03-27T17:10:06Z\",\"replicas\":1,\"node_selectors\":null,\"label_selectors\":{\"matchLabels\":{\"component\":\"litmusportal-server\"}},\"labels\":{\"component\":\"litmusportal-server\"},\"annotations\":null},{\"name\":\"subscriber\",\"uid\":\"9b64250d-225b-4973-8979-45079aea3b2c\",\"namespace\":\"litmus\",\"api_version\":\"\",\"cluster_name\":\"\",\"creation_timestamp\":\"2021-03-27T17:57:21Z\",\"replicas\":1,\"node_selectors\":null,\"label_selectors\":{\"matchLabels\":{\"app\":\"subscriber\"}},\"labels\":{\"app\":\"subscriber\"},\"annotations\":null},{\"name\":\"workflow-controller\",\"uid\":\"b69a8e11-8cc0-488b-908d-7fcd1515c4b4\",\"namespace\":\"litmus\",\"api_version\":\"\",\"cluster_name\":\"\",\"creation_timestamp\":\"2021-03-27T17:57:17Z\",\"replicas\":1,\"node_selectors\":{\"kubernetes.io/os\":\"linux\"},\"label_selectors\":{\"matchLabels\":{\"app\":\"workflow-controller\"}},\"labels\":null,\"annotations\":null}],\"statefulsets\":null,\"daemonsets\":null,\"deploymentconfiginfo\":null,\"rollouts\":null}]"
    }
  }
}`

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

File: litmus-portal/cluster-agents/subscriber/pkg/objects/objects.go [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+		podInfo := types.PodInfo{Name: pod.Name,
+			UID:               pod.UID,
+			Namespace:         pod.Namespace,
+			APIVersion:        pod.APIVersion,
+			ClusterName:       pod.ClusterName,
+			CreationTimestamp: pod.CreationTimestamp,
+			Containers:        pod.Spec.Containers,
+			NodeSelectors:     pod.Spec.NodeSelector,
+			Labels:            pod.GetLabels(),
+			//Annotations:     pod.GetAnnotations(),
```
Comment (arkajyotiMukherjee) [inline litmus-portal/cluster-agents/subscriber/pkg/objects/objects.go:153]: These comments....can these be removed or will it be required later?

Comment (amityt) [reply litmus-portal/cluster-agents/subscriber/pkg/objects/objects.go:153]: @arkajyotiMukherjee Yes, these are required. Was facing some issue while sending the data from subscriber with annotations. Will be updating it later.

File: litmus-portal/cluster-agents/subscriber/pkg/objects/objects.go [hunk 2]
```diff
@@ -0,0 +1,297 @@
+package objects
```
Comment (rajdas98) [inline litmus-portal/cluster-agents/subscriber/pkg/objects/objects.go:1]: I think we don't need these functions. There is already a generic function in k8s pkg to do the CRUD operations.

File: litmus-portal/cluster-agents/subscriber/pkg/types/kubeobject.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
+package types
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type PodInfo struct {
```
Comment (rajdas98) [inline litmus-portal/cluster-agents/subscriber/pkg/types/kubeobject.go:9]: We don't need to define all related k8s struct, you can directly use the struct from the official k8s repo.

**Review by rajdas98 — CHANGES_REQUESTED** [review]

File: litmus-portal/cluster-agents/subscriber/pkg/objects/objects.go (hunk 2 above)
Comment (rajdas98) [reply litmus-portal/cluster-agents/subscriber/pkg/objects/objects.go:1]: https://github.com/litmuschaos/litmus/blob/master/litmus-portal/cluster-agents/subscriber/pkg/k8s/operations.go

You can use/refactor operations.go acc to the usecase

**Comment by amityt** [comment]
Closing this PR since it was raised to the V2 branch which is already merged into master.

**Pull request closed without merging**